src/Nest/QueryDsl/Specialized/DistanceFeature/DistanceFeatureQuery.cs

```diff
@@ -0,0 +1,143 @@
+using System;
+using System.Runtime.Serialization;
+using Elasticsearch.Net.Utf8Json;
+using Elasticsearch.Net.Utf8Json.Internal;
+
+namespace Nest
+{
+	/// <summary>
+	/// Boosts the relevance score of documents closer to a provided origin date or point. For example, you can use this query to give
+	/// more weight to documents closer to a certain date or location.
+	/// You can use the distance_feature query to find the nearest neighbors to a location. You can also use the query in a bool
+	/// search’s should filter to add boosted relevance scores to the bool query’s scores.
+	/// </summary>
+	[JsonFormatter(typeof(DistanceFeatureQueryFormatter))]
+	[InterfaceDataContract]
+	public interface IDistanceFeatureQuery : IFieldNameQuery
+	{
+		/// <summary>
+		/// Date or point of origin used to calculate distances.
+		// If the field value is a date or date_nanos field, the origin value must be a date. Date Math, such as now-1h, is supported.
+		// If the field value is a geo_point field, the origin value must be a geopoint.
+		/// </summary>
+		[DataMember(Name = "origin")]
+		string Origin { get; set; }
```

Comment:
[Looks like this could be a `Union<DateMath, GeoLocation>`](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl-distance-feature-query.html#distance-feature-top-level-params)